src/serverConfig.ts

```diff
@@ -11,15 +12,16 @@ export const initServer = (): Express => {
 
     const dbClient = getClient();
     const userRepository = new UserRepository(dbClient);
+    const authenticationRepository = new AuthenticationRepository(dbClient);
 
     const helloController = new HelloController();
-    const userController = new UserController(userRepository);
+    const userController = new UserController(userRepository, authenticationRepository);
 
     app.post('/api/user/signup', (req: Request, res: Response) => {
         return userController.signUp(req, res);
     });
 
-    app.post('/api/user/signin', (req: Request, res: Response) => {
+    app.get('/api/user/signin', (req: Request, res: Response) => {
```

Comment:
it is wrong. it should be a post. you are sending sensitive data, and changing the application state.